### ISSUE TITLE
Can't map media buttons to global shortcuts to show main window (like "Launch mail" button)

### ISSUE THREAD

**Issue by MurzNN** [crow-translate/crow-translate#511]
**Bug description**
I have a multimedia keyboard with separate keys, that are recognized in Crow's key sequence input as: "Launch (1)", "Home page", "Launch mail", and I even can accept the shortcut.

But as result, all of them are not working for the "Show main window" action. Other combinations like default "Ctrl+Alt+C" work well.

Also, when I click "Ok" after configuring the shortcuts I see in the console output the line:
```
QHotkey: Unable to map shortcut to native keys. Key: Qt::Key_HomePage Modifiers: QFlags<Qt::KeyboardModifier>(NoModifier)
```

**Steps to reproduce**

1. Go to 'Settings » Shortcuts'
2. Configure "Launch mail" shortcut to the "Show main window" action.
3. Press "OK".
4. Quit and start the app to be sure.
5. Press the key. Nothing happens.

**Expected behavior**
The main window should be shown.

**Screenshots**
![image](https://user-images.githubusercontent.com/336662/216247387-a3bbffb3-d039-40a7-9319-f13156f5db24.png)

**Environment**

- OS: Ubuntu 
- Desktop environment: KDE
- Crow Translate version: 2.9.2

**Comment by MurzNN** [comment]
Also I've tried to make this action via dbus call like this command:
```
qdbus io.crow_translate.CrowTranslate /io/crow_translate/CrowTranslate/MainWindow open
```
It works well if the Crow window is hidden (closed or minimized), but does nothing if it is already opened, but in the background (below others).  

But the "Show main window"  shortcut works well for all the cases.

So, is there any way to bring the window to the foreground too via dbus command?

**Comment by Shatur** [comment]
Unfortunately, some shortcuts can't be mapped as global.
You could try to open an issue at [QHotkey](https://github.com/Skycoder42/QHotkey), it's the library we use.

**Comment by MurzNN** [comment]
Thanks for the explanation, have submitted here: https://github.com/Skycoder42/QHotkey/issues/81

**Comment by MurzNN** [comment]
And what about dbus, is it possible to activate the window via dbus call?

**Comment by Shatur** [comment]
Hm... This command works for me even if the window is in background:

> qdbus io.crow_translate.CrowTranslate /io/crow_translate/CrowTranslate/MainWindow open

**Comment by MurzNN** [comment]
Hmm, interesting, for me (Ubuntu 22.10, KDE 5.26.90, Frameworks 5.102.0). it just highlights the window in the taskbar like this:
![image](https://user-images.githubusercontent.com/336662/216289256-2bbf4c8f-6acc-42b1-bba1-65e71e710e57.png)

**Comment by Shatur** [comment]
Got it. I have a little bit older KDE. Will try to update later this week. But I afraid it could a KDE bug. Or maybe now they require more calls (bringing the window from the foregrounds depends on DE).
In the meanwhile, you could try play with some calls to show the main window here:
https://github.com/crow-translate/crow-translate/blob/69fd99d5c4c9ec6eaf227929d606c9891a5b0b30/src/mainwindow.cpp#L214-L227
[Docs](https://doc.qt.io/qt-5/qwindow.html).

**Comment by VolkMilit** [comment]
Are you sure your keyboard doesn't just send combinations and this "Launch mail" button really recognized as intended? I have an multimedia keyboard as well, it's an old Oklick 780L. For example calculator key actually sends Ctrl+Alt+C. If you're on X11, you can install xev (xorg-xev package) and check if keyboard really send multimedia key, or just emulates it.

For me, something like this shows in terminal:

```
KeyPress event, serial 40, synthetic NO, window 0x7200001,
    root 0x1e4, subw 0x0, time 3303979, (-121,233), root:(2030,676),
    state 0x10, keycode 37 (keysym 0xffe3, Control_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyPress event, serial 40, synthetic NO, window 0x7200001,
    root 0x1e4, subw 0x0, time 3303982, (-121,233), root:(2030,676),
    state 0x14, keycode 64 (keysym 0xffe9, Alt_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False
    ```

**Comment by MurzNN** [comment]
@VolkMilit Yes, I'm sure :)
Here are results from `xev`:
```
keycode 180 (keysym 0x1008ff18, XF86HomePage)
keycode 163 (keysym 0x1008ff19, XF86Mail)
keycode 148 (keysym 0x1008ff1d, XF86Calculator)
```

**Comment by VolkMilit** [comment]
Well, it's an QHotKey bug, then.

**Comment by Shatur** [comment]
I don't think so. I have the latest KDE and it works. Could it be a settings in Plasma?
